src/lib/eslint-naming.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * This is a direct copy and paste from eslint's github repository
+ *
+ * This will need updating if eslint changes their resolution function
+ * Eslint version as of 07/01/2019: 6.0.1
+ *
+ * https://github.com/eslint/eslint/blob/219aecb78bc646d44bad27dc775a9b3d3dc58232/lib/cli-engine/naming.js
```

Comment:
can this be required directly from eslint, instead of copy-pasted?

Reply:
I tried this method at first, but the file location changes between eslint 3, 4, 5 and 6. And I am unsure if it changes between different versions of those major releases as well. It doesn't look like it is exposed as an external API either.

Reply:
if it’s reachable, it’s part of the api.

It’s fine if we have to do try/catch requires to get at the function - that will ensure we get whatever matches the current version of eslint.

Reply:
How would you suggest handling eslint changing/moving file names between eslint minor releases? This plugin could break/work unexpectedly without warning.

Reply:
try/catch around require for each possible filename. It’s fine if eslint breaks us unexpectedly, that’d be a bug in a non-major.

Reply:
It doesn't look like [eslint v3 exports the functions](https://github.com/eslint/eslint/blob/3.x/lib/config/plugins.js) that we need here. What would you prefer I do? Copy the needed code or setup a fallback naming function? Or maybe it is time to drop eslint 3 support?

Honestly I still think copying the file is the best option as it works with all versions of eslint.

Reply:
Every function in that file is exported?

Reply:
The load function puts together the long and short names we need here. It doesn't return them, but instead directly requires the plugin here.

https://github.com/eslint/eslint/blob/3.x/lib/config/plugins.js#L109-L129

Reply:
it’s fine if for some versions of eslint we have to copypaste some of the code, but we should be maximally importing

Reply:
I force pushed (to keep PR commits clean, let me know if this isn't wanted in the future) an update. Let me know what you think

Reply:
Always preferred. Thanks, will take a look.